src/hooks/useConnectionStatus.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { useCallback, useContext, useEffect, useState } from 'react'
+
+export type ConnectionStatus = 'connected' | 'disconnected' | 'connecting' | 'no-internet'
+
+const ConnectionStatusContext = React.createContext<undefined | ConnectionStatusValue>(undefined)
+ConnectionStatusContext.displayName = 'ConnectionStatusContext'
+
+type ConnectionStatusValue = {
+  setNodeConnection: (connection: ConnectionStatus) => void
+  connectionStatus: ConnectionStatus
```

Comment:
I'd expect a separate status here for the internet/node connection. Could you walk me through how the status is determined here?

Reply:
1. First, the hook will check(and listen) if a user is connected to the internet. 
2. If a user is connected. it will set `connectionStatus` to `connecting`- which means basically `connecting to node`. If everything is fine with the node, it will set the state to `connected`, if not -`disconnected`
3. If a user doesn't have an internet connection it will set `connectionStatus` to `no-internet`. Now it will wait for reconnecting. If the user reconnects it will go to point `2` - set status to connecting and then it will check status of node

I thought it makes sense to keep this that way. If you cannot connect to the internet, it's impossible to determine if everything works on the node side, right?

Reply:
Yeah, I guess that mostly makes sense. Although I'm a bit worried on some details. For example, I guess it may be possible for the internet connection to go away for some short amount, like 1sec. So `offline`, `online` would get triggered. But I guess it's also possible for the node connection to not timeout in that time. Resulting in the end status of `connecting` and node not reporting getting connected since it think it always was. I guess we may have a better mental picture on this once we figure out the point above about `navigator.onLine`

Reply:
okay, I changed the approach. Now there are two separate statuses for the internet and for the node connection. 
If `navigator.online` will be true, the hook will be pinging google every three seconds to check if the user has a stable internet connection. 
If `navigator.online` will change to false, it will wait for three seconds, and then it will set the user as disconnected from the internet.